### PR TITLE
Cmake: Rewrite modules system

### PR DESCRIPTION
**Pull request by Winfidonarleyan**
##### CHANGES PROPOSED:
- Added new library `modules`. Now all modules will be in one library
image
- Now don't need to use a macro `AC_ADD_SCRIPT`. 

**From PR #985 by @AyaseCore**
- Added option `DISABLE_MODULES` which disables all modules
- For Windows Can disable and enable modules in Cmake gui
image
- Now in the config.cmake you can make a list of disabled modules
- - For example
```Cmake
# List disable modules
# Sets which modules need to be disabled.
# It looks like the following example.you need to remove the '#'
set(DISABLED_MODULES 
  test-module-2
  # mod-eluna-lua-engine
)
```

##### TESTS PERFORMED:
- [x] Build in VS 16 2019
- [x] Test in console win

##### HOW TO TEST THE CHANGES:
- Apply PR
- Clone 2 test modules
```shell
git clone https://github.com/Winfidonarleyan/test-module-1
git clone https://github.com/Winfidonarleyan/test-module-2
```
- Rebuild core
- Launch `worldserver`
- You should see something similar
```
Kargatum_Test_Module_2 is enable!
Kargatum_Test_Module_1 is enable!
AC>Starting up Auction House Listing thread...
```

##### KNOWN ISSUES AND TODO LIST:
- [x] Build in Travis default
- [ ] Build in Travis witch 1 disable modules
- [ ] Build in Travis witch Disable all modules

##### Target branch(es): Master

### CODE REVIEW

File: modules/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  CollectSourceFiles(
+    ${CMAKE_SOURCE_DIR}/${subdir_rel}/
+    PRIVATE_SOURCES)
+
+  set(MODULES_STAT_SRCS
+    ${MODULES_STAT_SRCS}
+    ${PRIVATE_SOURCES})
+ENDFOREACH()
+
+# Группировка сорцов
```
Comment (FrancescoBorzi) [inline modules/CMakeLists.txt]: english please :)

Comment (Winfidonarleyan) [reply modules/CMakeLists.txt]: ah, yes

**Comment by FrancescoBorzi** [comment]
@Winfidonarleyan why closing?

**Comment by Winfidonarleyan** [comment]
Pr is useless

**Pull request closed without merging**